kubernetes/resource_kubernetes_job_v1.go

```diff
@@ -173,6 +230,31 @@ func resourceKubernetesJobV1Update(ctx context.Context, d *schema.ResourceData,
 		return diag.FromErr(err)
 	}
 
+	// Attempt to get the Job
+	_, err = conn.BatchV1().Jobs(namespace).Get(ctx, name, metav1.GetOptions{})
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			// Job is missing; check TTL
+			ttlAttr := d.Get("spec.0.ttl_seconds_after_finished")
+			ttlStr, _ := ttlAttr.(string)
+			ttlInt, err := strconv.Atoi(ttlStr)
+			if err != nil {
+				ttlInt = 0
+			}
+
+			if ttlInt >= 0 {
+				// Job was deleted due to TTL nothing to update
+				log.Printf("[INFO] Job %s not found but ttl_seconds_after_finished = %v; nothing to update", d.Id(), ttlInt)
+				return nil
+			}
+
+			// Job was deleted unexpectedly; return an error
+			return diag.Errorf("Job %s not found; cannot update because it has been deleted", d.Id())
+		}
+		return diag.Errorf("Error retrieving Job: %s", err)
+	}
```

Comment:
I don't think you need this logic either. In the read function it's going to check if the job exists and then set the ID to `""` if it doesn't which will force recreation and so the update function wont get called. 

I tried commenting this code out and all your tests still pass too.